client/src/routes/widgetModule/WidgetModule.tsx

```diff
@@ -95,6 +125,188 @@ export const WidgetModule: React.FC<Props> = observer(({ domainName }) => {
     setFormFields({ ...formFields, widgetValue: '' })
   }
 
+  const createAlias = async (alias: string, forward: string) => {
+    setProcessStatus({
+      type: ProcessStatusTypes.PROGRESS,
+      render: (
+        <BaseText>
+          {walletStore.isMetamaskAvailable
+            ? 'Waiting for a transaction to be signed'
+            : 'Sign transaction on mobile device'}
+        </BaseText>
+      ),
+    })
+
+    try {
+      const numAlias = await rootStore.easClient.getNumAlias(domainName)
+      const maxAlias = await rootStore.easClient.maxNumAlias()
+      const publicAliases = await rootStore.easClient.getPublicAliases(
+        domainName
+      )
+      console.log('### maxNum', numAlias)
+      console.log('### numAlias', numAlias)
+      console.log('### publicAliases', publicAliases)
+
+      if (numAlias >= maxAlias) {
+        setProcessStatus({
+          type: ProcessStatusTypes.PROGRESS,
+          render: (
+            <BaseText>
+              {walletStore.isMetamaskAvailable
+                ? 'Waiting for a transaction to be signed'
+                : 'Sign transaction on mobile device'}
+            </BaseText>
+          ),
+        })
+
+        const removingAlias = publicAliases[0]
+
+        const newPublicAliases = publicAliases.filter(
+          (item) => item !== removingAlias
+        )
+
+        const setResult = await rootStore.easClient.setPublicAliases({
+          domainName,
+          aliases: newPublicAliases,
+          onTransactionHash: () => {
+            setProcessStatus({
+              type: ProcessStatusTypes.PROGRESS,
+              render: <BaseText>Waiting for transaction confirmation</BaseText>,
+            })
+          },
+        })
+
+        if (setResult.error) {
+          throw setResult.error
+        }
+
+        const delResult = await rootStore.easClient.deactivate({
+          domainName,
+          alias: removingAlias,
+          onTransactionHash: () => {
+            setProcessStatus({
+              type: ProcessStatusTypes.PROGRESS,
+              render: <BaseText>Waiting for transaction confirmation</BaseText>,
+            })
+          },
+        })
+
+        if (delResult.error) {
+          throw delResult.error
+        }
+
+        const { success, error } = await easServerClient.deactivate(
```

Comment:
what if this fails? you end up with the alias removed on-chain, but not on server, so the next attempt would just fail at the on-chain step (because you can't deactivate an alias that is already gone) and the user would get stuck in error loop

Reply:
yes, i understand already

What should we do? Since if it can happen, it will happen. (network error, or user mistakes)

Reply:
I use check-alias server api to check consistency between server alias state and on-chain alias state. You can see some implementations here https://github.com/harmony-one/eas/blob/a226a3a3d0f8375d702eab1abfbe9e42ed86a538/client/src/Home.tsx#L137